### PR TITLE
GH-168: Upgrade to SI-4.3.6 and others

### PR DESCRIPTION
**Pull request by artembilan**
Fixes GH-168 (https://github.com/spring-projects/spring-integration-extensions/issues/168)

### CODE REVIEW

**Comment by garyrussell** [comment]
Travis was not enabled for this repo; I have enabled it now; if you re-push with a new hash, we should get one.

File: spring-integration-smb/build.gradle [hunk 1]
```diff
@@ -1,43 +1,62 @@
...
+				mavenBom "io.spring.platform:platform-bom:${platformVersion}"
+			}
+		}
+	}
+}
+
+compileJava {
+	sourceCompatibility = 1.7
+	targetCompatibility = 1.7
 }
```
Comment (garyrussell) [inline spring-integration-smb/build.gradle]: 1.7?

**Comment by garyrussell** [comment]
Actually, I am not sure travis can work because each extension has its own build.

**Comment by garyrussell** [comment]
Maybe put `.travis` at the top level and cd into the subproject?

Or add a top-level gradlw?

**Comment by artembilan** [comment]
Sorry for delay and thank you for merging this one.
Well, since this project hasn't been released yet and it is an extension I think that shouldn't hurt to stick with Java 7 as minimum. Java is EOL already, even if SI 4.3.x is still with that 😄 

Regarding Travis, well, let me push some other PR an will see. At least I see Travis config in the Hazelcast directory. If it is there, that might be a case that Travis works.

I see that CI plan has to be fixed as well.

**Pull request merged**